decode.go

```diff
@@ -1279,7 +1317,29 @@ func (d *decoder) parse(skipSelfDescribedTag bool) (interface{}, error) { //noli
 		return nValue, nil
 
 	case cborTypeByteString:
-		return d.parseByteString(), nil
+		switch d.dm.defaultByteStringType {
+		case nil, reflect.TypeOf([]byte(nil)):
```

Comment:
nit: maybe create a global var for `reflect.TypeOf([]byte(nil))` so it can be reused.